API/Controllers/ProjectController.cs

```diff
@@ -1058,6 +1104,176 @@ public async Task<IActionResult> UnlinkInstitutionFromProjectAsync(int projectId
 
             return Ok();
         }
+
+        /// <summary>
+        ///     Categorize a project
+        /// </summary>
+        /// <param name="projectId"></param>
+        /// <param name="categoryId"></param>
+        /// <response code="200">Category was added to the project.</response>
+        /// <response code="401">User is not authorized to add the category to the project.</response>
+        /// <response code="404">Project or category was not found.</response>
+        /// <response code="409">Project is already categorized with the category.</response>
+        /// <returns></returns>
+        [HttpPost("category/{projectId}/{categoryId}")]
+        [Authorize]
+        [ProducesResponseType(typeof(ProjectResourceResult), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status409Conflict)]
+        public async Task<IActionResult> ProjectAddCategory(int projectId, int categoryId)
+        {
+            Project project = await projectService.FindAsync(projectId)
+                                                  .ConfigureAwait(false);
+            if(project == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed to categorize the project.",
+                    Detail = "The project could not be found in the database.",
+                    Instance = "1C8D069D-E6CE-43E2-9CF9-D82C0A71A292"
+                };
+                return NotFound(problem);
+            }
+
+            Category category = await categoryService.FindAsync(categoryId)
+                                      .ConfigureAwait(false);
+
+            if(category == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed to categorize the project.",
+                    Detail = "The category could not be found in the database.",
+                    Instance = "93C6B5BD-EC14-482A-9907-001C888F3D3F"
+                };
+                return NotFound(problem);
+            }
+
+            User user = await HttpContext.GetContextUser(userService)
+                                         .ConfigureAwait(false);
+            bool isAllowed = await authorizationHelper.UserIsAllowed(user,
+                                                                     nameof(Defaults.Scopes.AdminProjectWrite),
+                                                                     nameof(Defaults.Scopes.InstitutionProjectWrite),
+                                                                     project.UserId);
+
+            if(project.UserId != user.Id && !isAllowed)
```

Comment:
Unauthorized should return if one or both conditions are false, right?

```suggestion
            if(!(project.UserId == user.Id || isAllowed))
```